internal/parser/parser.go

```diff
@@ -0,0 +1,47 @@
+package parser
+
+// Parser defines the SDL parser
+type Parser struct {
+	parts   []Part // Array of parts. Our simple AST
+	input   string // statement to be parsed
+	parsed  int    // position of the last parsed chunk
+	skipped int    // current position of the parser
+}
+
+// NewParser creates a new parser
+func NewParser() *Parser {
+	return &Parser{}
+}
+
+// init initializes the parser
+func (p *Parser) init(input string) {
+	p.parts = nil
+	p.input = input
+	p.parsed = 0
+	p.skipped = 0
+}
+
+func (p *Parser) Parse(input string) (*ParsedExpr, error) {
+	p.init(input)
+	return nil, nil
+}
+
+// Part defines a simple interface for all the different parts that
+// compose a ParsedExpr
+type Part interface {
```

Comment:
I'd declare this at the top ahead of it's first usage in `Parser`

Reply:
Done!